src/nrtk_explorer/app/transforms.py

```diff
@@ -182,9 +189,12 @@ def on_current_num_elements_change(self, current_num_elements, **kwargs):
         with open(self.state.current_dataset) as f:
             dataset = json.load(f)
         ids = [img["id"] for img in dataset["images"]]
-        return self.on_selected_images_change(ids[:current_num_elements])
+        return self.set_source_images(ids[:current_num_elements])
 
-    def on_selected_images_change(self, selected_ids):
+    async def _set_source_images(self, selected_ids):
+        # We need to yield twice for the loading_images=True to commit to the trame state
```

Comment:
no loading_images here.

Reply:
Confusing.  I refactored so manipulation of `self.state.loading_images` is  closer to this comment.

Reply:
oops that was my bad :facepalm: , I did not scrolled down, I thought it was copy-pasta :angel: